File: python/ex1.py
import numpy as np
from libs import solve_lyapunov, h2_norm

A = np.array(
    [
        [0, 1],
        [-10, -1],
    ]
)
B = np.array(
    [
        [0],
        [1],
    ]
)

C = np.array([[1, 0]])

print(solve_lyapunov(A, C.T @ C))

print(h2_norm(A, B, C))

File: cpp/ex1.cpp
#include <iostream>
#include "robust_control.h"
#include <Eigen/Eigen>

using namespace std;
using namespace Eigen;

int main(int argc, char const **argv)
{
    MatrixXd A{
        {0, 1},
        {-10, -1},
    };
    MatrixXd B{
        {0},
        {1},
    };
    MatrixXd C{
        {1, 0},
    };
    auto Q = solveLyapunov(A, C.transpose() * C);
    cout << Q << endl;
    cout << h2_norm(A, B, C) << endl;
    return 0;
}
